packages/react-query-devtools/src/devtools.tsx

```diff
@@ -377,12 +377,18 @@ export function ReactQueryDevtools({
 const useSubscribeToQueryCache = <T,>(
   queryCache: QueryCache,
   getSnapshot: () => T,
+  skip: boolean = false,
 ): T => {
   return useSyncExternalStore(
     React.useCallback(
-      (onStoreChange) =>
-        queryCache.subscribe(notifyManager.batchCalls(onStoreChange)),
-      [queryCache],
+      (onStoreChange) => {
+        if (!skip)
+          return queryCache.subscribe(notifyManager.batchCalls(onStoreChange))
+        return () => {
+          return
+        }
```

Comment:
I think this could just be `return undefined`

so I would do, to avoid the negated condition:

```
return skip ? undefined : queryCache.subscribe(notifyManager.batchCalls(onStoreChange))
```

Reply:
Tried it

```ts
useSyncExternalStore<T>(subscribe: (onStoreChange: () => void) => () => void, getSnapshot: () => T, getServerSnapshot?: (() => T) | undefined): T
```

![Screenshot 2022-09-11 at 23 20 18](https://user-images.githubusercontent.com/69076988/189541844-f94abec6-4044-4f04-b2c0-30b801116a80.png)
![Screenshot 2022-09-11 at 23 20 34](https://user-images.githubusercontent.com/69076988/189541857-678cbe9a-91c4-4092-b221-db45181f14de.png)

Reply:
oh i didn't know that, I thought the API is the same as the one from useEffect. nevermind then 🙈